openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -222,7 +237,7 @@ def get_cohort(user, course_key, assign=True):
 
     user.course_groups.add(cohort)
 
-    return cohort
+    return request_cache.data.setdefault(cache_key, cohort)
```

Comment:
The data in stored in a threading.local() object which is cleared in `process_request` and `process_response` functions of `request_cache.middleware.RequestCache`. Since the management commands/celery tasks do not use the response-request cycle, these middleware functions will not get called.

1. I am not sure if celery creates a new thread for each task otherwise this cache would continue to grow.
2. The problem with manually flushing the cache for every user is that it seems the cache also caches course data, `metadata_inheritance` etc which would get flushed too and would need to be refetched.


https://github.com/edx/edx-platform/blob/master/common/djangoapps/request_cache/middleware.py